### PR TITLE
Update telemetry dependency to stable ~> 1.0

### PR DESCRIPTION
**Pull request by bmuller**
There [were no changes in the Telemetry 1.0.0 release](https://github.com/beam-telemetry/telemetry/blob/main/CHANGELOG.md#100) - but it does represent the first "stable" version of the API.

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -60,7 +60,7 @@ defmodule Dataloader.Mixfile do
 
   defp deps do
     [
-      {:telemetry, "~> 0.4"},
+      {:telemetry, "~> 1.0"},
```
Comment (benwilson512) [inline mix.exs:63]:
```suggestion
      {:telemetry, "~> 1.0 or ~> 0.4"},
```

**Review by benwilson512 — CHANGES_REQUESTED** [review]
Let's make this an `or` so that it maintains compatibility.

**Comment by edennis** [comment]
Any chance at a version bump to include this?

**Comment by krainboltgreene** [comment]
Jumping in here because phoenix 1.6 released with telemetry 1.0.

**Pull request merged**